doc/nrf/ug_memfault.rst

```diff
@@ -0,0 +1,208 @@
+
+.. note::
+   Memfault has been integrated in |NCS| since the v1.6.0 release.
+
+Prerequisites
+*************
+
+Before you start the |NCS| integration with Memfault, make sure that the following prerequisites are completed:
+
+   * :ref:`Setup of nRF Connect SDK <getting_started>` environment.
+   * :ref:`Setup of nRF9160 DK <ug_nrf9160_gs>`.
+   * Sign up in the `Memfault registration page`_ and `create a new project in Memfault`_.
+     You will be directed to the Integration guide for Memfault.
+     You receive the ``project key`` here to set in the :kconfig:option:`CONFIG_MEMFAULT_NCS_PROJECT_KEY` Kconfig option.
+     After registration, you can connect up to 100 devices for free.
+
+Solution architecture
+*********************
+
+The SDK provides functionality to collect debug information from your devices in the form of coredumps, error traces, metrics, logs, and more.
+Memfault saves the data to a configurable storage, after which it is broken into chunks and forwarded at intervals or application-specific times to Memfault's cloud solution for further analysis.
+Communication with the Memfault Cloud is handled by APIs available in `Memfault-SDK`_, while the integration code of the SDK in |NCS| provides some additional functionality.
+The Memfault cloud reassembles the data before analyzing and deduplicates the data.
+The overall fleet health of the issues can be examined through the dashboard of the Memfault web application.
+See the `Memfault MCU Guide`_ for more information.
+
+.. figure:: /images/Memfault_architecture.svg
+   :alt: Memfault Solution architecture
+
+   Memfault Solution architecture
+
+.. _using_memfault:
+
+Integration overview
+********************
+
+The SDK version is included in the `west manifest file`_ in |NCS| and is automatically downloaded when running ``west update``.
+By default, it is downloaded to ``nRF/modules/memfault-firmware-sdk/``.
+
+Complete the following steps to integrate Memfault into |NCS|:
+
+1. :ref:`Configure Memfault <ug_memfault_config>`
+#. :ref:`Track errors <ug_memfault_error>`
+#. :ref:`Collect metrices <ug_memfault_metrics>`
+
+.. _ug_memfault_config:
+
+Configuring Memfault
+====================
+
+To include Memfault in your build, add the following Kconfig option in your :file:`prj.conf` file:
+
+.. code-block:: console
+
+   CONFIG_MEMFAULT=y
+   CONFIG_MEMFAULT_NCS_PROJECT_KEY=""
+
+The APIs in the Memfault SDK can then be linked into your application.
+You can get your project key after signing up with Memfault.
+
+.. note::
+   In the :ref:`memfault_sample` sample folder, the aforementioned Memfault configurations are already included.
+   Make sure to add your project key in the :kconfig:option:`CONFIG_MEMFAULT_NCS_PROJECT_KEY` Kconfig option.
+
+You can also add configuration using multiple overlay files, ``-DOVERLAY_CONFIG`` must be set to a list of overlay configurations.
+See the following example on how to set an overlay configuration for Memfault integration:
```

Comment:
```suggestion
You can also add the configurations using a Kconfig configuration fragment file.
See the following example on how to set an overlay configuration for Memfault integration:
```

Reply:
I know we use overlay a lot for these files, but they're now calles Kconfig configuration fragment files according to this one: https://docs.zephyrproject.org/3.2.0/develop/application/index.html#important-build-system-variables